std/haxe/http/HttpJs.hx

```diff
@@ -74,7 +74,7 @@ class HttpJs extends haxe.http.HttpBase {
 			if (s != null && s >= 200 && s < 400) {
 				req = null;
 				success(Bytes.ofData(r.response));
-			} else if (s == null) {
+			} else if (s == null || s == 0) {
```

Comment:
This should be changed to `|| (s == 0 && r.response == null)`
Or add `responseBytes = r.response != null ? Bytes.ofData(r.response) : null;` to the body of this `if`
That is needed to not lose response body in case browsers set `s = 0` in other weird edge cases (like local requests).

Reply:
I would prefer the first, because this is the case for the error message - you can't get a response if the browser is unable to resolve the url. Then the other weird cases would be covered in the `default` branch with a less explicit error message.